lib/Doctrine/Persistence/AbstractManagerRegistry.php

```diff
@@ -173,20 +174,39 @@ public function getManagerForClass($class)
                 return null;
             }
 
-            $class = $parentClass->getName();
+            $className = $parentClass->getName();
         }
 
         foreach ($this->managers as $id) {
             $manager = $this->getService($id);
 
-            if (! $manager->getMetadataFactory()->isTransient($class)) {
+            if (! $manager->getMetadataFactory()->isTransient($className)) {
                 return $manager;
             }
         }
 
         return null;
     }
 
+    /**
+     * @param string $className
+     *
+     * @psalm-return class-string
+     */
+    private function getRealClassName($className): string
+    {
+        // Check for namespace alias
+        if (strpos($className, ':') !== false) {
+            [$namespaceAlias, $simpleClassName] = explode(':', $className, 2);
+
+            /** @psalm-var class-string */
+            return $this->getAliasNamespace($namespaceAlias) . '\\' . $simpleClassName;
+        }
+
+        /** @psalm-var class-string */
+        return $className;
```

Comment:
Maybe an assertion for class_exists could be used in the future instead of overriding? In a future version to avoid BC breaks maybe?